Config.uk

```diff
@@ -0,0 +1,19 @@
+menuconfig LIBXZ
+    bool "xz-utils - LZMA compression library and XZ support"
+    select LIBPOSIX_SYSINFO
+    select LIBINTEL_INTRINSICS if ARCH_X86_64
+    default n
+
+if LIBXZ
+    config LIBXZ_MAIN
+    bool "Provide 'xz' utility as main"
+    depends on HAVE_LIBC
+    select LIBPOSIX_EVENT
+    default n
+
+    config LIBXZ_THREAD
+    bool "Enable threading support"
+    depends on LIBMUSL_THREAD
```

Comment:
I kind of do not like this, since we allow other libcs by using `HAVE_LIBC`, but enforce the use of `musl` with `LIBMUSL_THREAD`.
I don't really see a way out though, so I wouldn't block on this unless you can see a better way to do it.

Reply:
We could use `HAVE_PTHREAD`, but that can't account for how much pthread support a lib has (see longer response https://github.com/unikraft/lib-littlecms/pull/1). In this case I'm also tempted to remove this option until (1) SMP support lands, so it actually makes sense and (2) we sort out the libc/pthread situation.